### DISCUSSION TITLE
Frequency Calibration

### DISCUSSION THREAD

**Discussion by daelsc** [emporia-vue-local/esphome#190] [Q&A] [answered]
The reported Phase A frequency is 1Hz lower than measure with a Fluke multimeter.  (59hz in ESPHome vs. 60Hz on Fluke).  Is there a way to calibrate frequency like we are able to calibrate voltage?

It is also surprising to me that the frequency is exactly 1Hz too low, might this be a bug in the calculation?  Are other folks seeing exactly 60Hz (or 50Hz) reported?

**Comment by clowrey** [comment]
Use a sensor filter in ESPhome to add +1 to the value.

**Comment by flaviut** [comment, marked as answer]
Some folks are seeing too high or just right: https://github.com/emporia-vue-local/esphome/pull/88. I personally do not capture this information since I don't see the point.

To fix this, I'd suggest [using `calibrate_linear` filter](https://esphome.io/components/sensor/index.html?highlight=filter#calibrate-linear) instead of adding 1: `calibrate_linear: ["0 -> 0", "59 -> 60"]`. Hopefully I got the syntax right there.

Keep in mind the Vue has a resolution of maybe ±0.5 here. It is still very imprecise even if you calibrate it to be accurate.

**Reply by daelsc** [reply]
implemented the calibrate_linear filter.  Works like a charm.  thank you both for the help.

**Reply by daelsc** [reply]
I'll add that accurate frequency measurement is helpful in off-grid battery/solar applications that use frequency shifting to adjust inverter output.  I love my Vue2 with ESPHome.  Thanks so much for the great work!
